subprojects/ide-native/src/integTest/groovy/org/gradle/ide/xcode/XcodeCppExternalSourceDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,141 @@
+        settingsFile << """
+            sourceControl {
+                vcsMappings {
+                    withModule("org.test:greeter") {
+                        from vcs(GitVersionControlSpec) {
+                            url = "${repo.url}"
+                        }
+                    }
+                }
+            }
+        """
+        buildFile << """
+            apply plugin: 'cpp-application'
+            apply plugin: 'xcode'
+            group = 'org.gradle'
+            version = '2.0'
+
+            dependencies {
+                implementation "org.test:greeter:latest.integration"
+            }
+        """
+        fixture.main.writeToProject(testDirectory)
+
+        when:
+        succeeds 'xcode'
+
+        then:
+        executedAndNotSkipped(":xcodeProject", ":xcodeProjectWorkspaceSettings", ":xcodeSchemeAppExecutable", ":xcode")
+        rootXcodeWorkspace.contentFile.assertHasProjects("${rootProjectName}.xcodeproj")
+
+        def appProject = xcodeProject("${rootProjectName}.xcodeproj").projectFile
+        appProject.indexTarget.getBuildSettings().HEADER_SEARCH_PATHS == toSpaceSeparatedList(file('src/main/headers'), checkoutDir(repo.name, commit.id.name, repo.id).file('src/main/public'))
+    }
+
+    def "adds source dependencies Xcode project of main component to Xcode workspace"() {
+        def fixture = new CppAppWithLibrary()
+
+        given:
+        buildTestFixture.withBuildInSubDir()
+        depProject = singleProjectBuild("greeter") {
+            buildFile << """
+                apply plugin: 'cpp-library'
+                apply plugin: 'xcode'
+            """
+            fixture.greeter.writeToProject(it)
+        }
+        def commit = repo.commit('initial commit', GFileUtils.listFiles(file('greeter'), null, true))
+
+        and:
+        settingsFile << """
+            sourceControl {
+                vcsMappings {
+                    withModule("org.test:greeter") {
+                        from vcs(GitVersionControlSpec) {
+                            url = "${repo.url}"
+                        }
+                    }
+                }
+            }
+        """
+        buildFile << """
+            apply plugin: 'cpp-application'
+            apply plugin: 'xcode'
+            group = 'org.gradle'
+            version = '2.0'
+
+            dependencies {
+                implementation "org.test:greeter:latest.integration"
+            }
+        """
+        fixture.main.writeToProject(testDirectory)
+
+        when:
+        succeeds 'xcode'
+
+        then:
+        executedAndNotSkipped(":xcodeProject", ":xcodeProjectWorkspaceSettings", ":xcodeSchemeAppExecutable", ":xcode",
+            ":greeter:xcodeProject", ":greeter:xcodeProjectWorkspaceSettings", ":greeter:xcodeSchemeGreeterSharedLibrary")
+        rootXcodeWorkspace.contentFile.assertHasProjects("${rootProjectName}.xcodeproj", "${checkoutRelativeDir(repo.name, commit.id.name, repo.id)}/greeter.xcodeproj")
```

Comment:
we don't want the `greeter` library to be included in the workspace as a project, but as binaries.